### PR TITLE
Actions -- updated documentation

### PR DESCRIPTION
**Pull request by CarlosNZ**
Have updated the documentation for all the current Actions, with a bit about Core Actions.

### CODE REVIEW

File: documentation/List-of-Action-plugins.md [hunk 1]
```diff
@@ -2,59 +2,256 @@
...
+
+- _Action Code:_ **`generateReviewAssignments`**
+
+| Input parameters<br />(\*required) <br/> | Output properties                 |
+| ---------------------------------------- | --------------------------------- |
+| `applicationId`\*                        | `reviewAssignmentIds`             |
+| `reviewId`                               | `reviewAssignmentAssignerJoins`   |
+| `templateId`\*                           | `reviewAssignmentAssignerJoinIds` |
+| `stageId`\*                              | `currentReviewLevel`              |
+| `stageNumber`\*                          | `nextReviewLevel`                 |
```
Comment (nmadruga) [inline documentation/List-of-Action-plugins.md:160]: `templateId`, `stageId`, `stageNumber` aren't passed as props

File: documentation/List-of-Action-plugins.md [hunk 2]
```diff
@@ -2,59 +2,256 @@
...
+
+- _Action Code:_ **`updateReviews`**
+
+| Input parameters<br />(\*required) <br/> | Output properties |
+| ---------------------------------------- | ----------------- |
+| `applicationId`\*                        | `updatedReviews`  |
+
+## Core Actions
+
+There are certain Actions that _must_ run on particular to facilitate a standard application/review workflow process. We have called these **Core Actions**, and they have been collected in a single "core_actions.js" file (for insertion into database via GraphQL). Each template (other than "User Registration") has this block slugged into it as a template literal (`${coreActions}`) in its Action insertion block, before any Actions that are specific to that template.
```
Comment (nmadruga) [inline documentation/List-of-Action-plugins.md:216]: `run on particular` I'm not familiar with this expression. Is this correct?

File: documentation/List-of-Action-plugins.md [hunk 3]
```diff
@@ -2,59 +2,256 @@
...
+- Nothing yet (To-do?)
+
+#### On Review Create
+
+- Change Status (to "Draft")
+
+#### On Review Submit:
+
+- Change Status (to "Submitted")
+- Trim Responses
```
Comment (nmadruga) [inline documentation/List-of-Action-plugins.md:250]: - Increment stage (after last level reviewer conforms)

**Review by nmadruga — APPROVED** [review]
Oh! This is great. Very nice to keep things tidy with an updated version of the list of actions.
Have made a few comments only but looks very good to me. So considering approved!

File: database/insertData/core_actions.js [hunk 4]
```diff
@@ -60,6 +60,7 @@ exports.coreActions = `
     # 1 - change status to submitted
     # 2 - trim responses
     # 3 - generate review assignments
+    # 4 - update review statuses
```
[the remark this answers is not in the document]
Comment (CarlosNZ) [reply database/insertData/core_actions.js:63]: I just merged from master and there's no problems. That PR doesn't actually change `core_actions`. Did you mean a different one?

File: documentation/Database-Schema-Review-And-Consolidations.md [hunk 5]
```diff
@@ -59,7 +59,11 @@ These records are created upon assignment, and should only be created for level
 Either a review of applications questions (base review, level 1) or review of review (consolidation, level > 1). See diagram (responses flow). Always created by front end, but duplicates are trimmed on back end.
 Can be used to agree or disagree on overall review_decision.
 
-A few trick bits:
+## Review Assignment Assigner Join
+
+A collection of users who have permission to assign other users (reviewers) to applications. These records are generated upon submission of any application or review: after review assignments are generated, each new review assignment is joined with the users who have "Assign" permissions for those applications. (This happens as part of the **generateReviewAssignments** Action)
```
[the remark this answers is not in the document]
Comment (CarlosNZ) [reply documentation/Database-Schema-Review-And-Consolidations.md:64]: 'assigner users who have "Assign" permissions' -- doesn't that sound a bit redundant? I'd either say "Assigners" OR "users with Assign permissions". Both sounds clunky to me.

File: documentation/List-of-Action-plugins.md [hunk 6]
```diff
@@ -2,59 +2,256 @@
...
+
+| Input parameters<br />(\*required) <br/> | Output properties                 |
+| ---------------------------------------- | --------------------------------- |
+| `applicationId`\*                        | `reviewAssignmentIds`             |
+| `reviewId`                               | `reviewAssignmentAssignerJoins`   |
+| `templateId`\*                           | `reviewAssignmentAssignerJoinIds` |
+| `stageId`\*                              | `currentReviewLevel`              |
+| `stageNumber`\*                          | `nextReviewLevel`                 |
+
+Note: if a `reviewId` is supplied, the Action will generate review assignments for that _review_. If not, they will be generated for the _application_ (i.e. level 1)
```
[the remark this answers is not in the document]
Comment (CarlosNZ) [reply documentation/List-of-Action-plugins.md:162]: Yeah, cool, I have summarised that info in slightly different wording.

File: documentation/List-of-Action-plugins.md (hunk 2 above)
Comment (CarlosNZ) [reply documentation/List-of-Action-plugins.md:216]: Sorry, missing a word. Should be "run on particular events"

File: documentation/List-of-Action-plugins.md (hunk 3 above)
Comment (CarlosNZ) [reply documentation/List-of-Action-plugins.md:250]: Okay, added.

**Comment by CarlosNZ** [comment]
Thanks for the review @nmadruga . Changes made 0b5e6fef if you want to take a look.

File: documentation/Database-Schema-Review-And-Consolidations.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (nmadruga) [reply documentation/Database-Schema-Review-And-Consolidations.md:64]: No problem to leave as is. It just seemed missing to me.

File: database/insertData/core_actions.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (nmadruga) [reply database/insertData/core_actions.js:63]: Yes, sorry haha was meaning to say this PR #280

File: documentation/List-of-Action-plugins.md (hunk 2 above)
Comment (nmadruga) [reply documentation/List-of-Action-plugins.md]: Cool. Thanks : )

File: documentation/List-of-Action-plugins.md [hunk 7]
```diff
@@ -2,59 +2,259 @@
...
+
+| Input parameters<br />(\*required) <br/> | Output properties                 |
+| ---------------------------------------- | --------------------------------- |
+| `applicationId`\*                        | `reviewAssignmentIds`             |
+| `reviewId`                               | `reviewAssignmentAssignerJoins`   |
+| `templateId`\*                           | `reviewAssignmentAssignerJoinIds` |
+|                                          | `currentReviewLevel`              |
+|                                          | `nextReviewLevel`                 |
+
+Note: if a `reviewId` is not supplied, the Action will generate review assignment records for Stage 1, level 1 review. If it is present, it'll generate records for the _next_ review level (either in the same Stage or level 1 of the next Stage)
```
Comment (nmadruga) [inline documentation/List-of-Action-plugins.md:164]: Great!

File: documentation/List-of-Action-plugins.md [hunk 8]
```diff
@@ -2,59 +2,259 @@
...
+
+_TO-DO: decide what to do with records that exist, but no longer should be (e.g. the reviewer has been removed/permissions revoked)_
+
+- _Action Code:_ **`generateReviewAssignments`**
+
+| Input parameters<br />(\*required) <br/> | Output properties                 |
+| ---------------------------------------- | --------------------------------- |
+| `applicationId`\*                        | `reviewAssignmentIds`             |
+| `reviewId`                               | `reviewAssignmentAssignerJoins`   |
+| `templateId`\*                           | `reviewAssignmentAssignerJoinIds` |
```
Comment (nmadruga) [inline documentation/List-of-Action-plugins.md:160]: `templateId` can also be removed

**Review by nmadruga — APPROVED** [review]

File: documentation/List-of-Action-plugins.md (hunk 8 above)
Comment (CarlosNZ) [reply documentation/List-of-Action-plugins.md]: Cool. I also went an combined the Notes and added a bit more in the "TO-DO" section.

File: database/insertData/core_actions.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (CarlosNZ) [reply database/insertData/core_actions.js:63]: Cool -- well I'm about to merge this, so I think I'll beat you to it 😜

**Pull request merged**